core/domain/stats_domain.py

```diff
@@ -123,23 +145,117 @@ def validate(self):
+        self.classification_categorization = classification_categorization
+        self.params = params
+        self.session_id = session_id
+        self.time_spent_in_sec = time_spent_in_sec
+        self.rule_spec_str = rule_spec_str
+        self.answer_str = answer_str
+
+    def to_dict(self):
+        submitted_answer_dict = {
+            'answer': self.normalized_answer,
+            'interaction_id': self.interaction_id,
+            'answer_group_index': self.answer_group_index,
+            'rule_spec_index': self.rule_spec_index,
+            'classification_categorization': self.classification_categorization,
+            'params': self.params,
+            'session_id': self.session_id,
+            'time_spent_in_sec': self.time_spent_in_sec
+        }
+        if self.rule_spec_str is not None:
+            submitted_answer_dict['rule_spec_str'] = self.rule_spec_str
+        if self.answer_str is not None:
+            submitted_answer_dict['answer_str'] = self.answer_str
+        return submitted_answer_dict
 
-        # Note: There is no need to validate content of answers_list here
-        # because each answer is validated before it is appended to
-        # answers_list (which is faster than validating whole answers_list
-        # each time a new answer is recorded).
+    @classmethod
+    def from_dict(cls, submitted_answer_dict):
+        return cls(
+            submitted_answer_dict['answer'],
+            submitted_answer_dict['interaction_id'],
+            submitted_answer_dict['answer_group_index'],
+            submitted_answer_dict['rule_spec_index'],
+            submitted_answer_dict['classification_categorization'],
+            submitted_answer_dict['params'],
+            submitted_answer_dict['session_id'],
+            submitted_answer_dict['time_spent_in_sec'],
+            rule_spec_str=submitted_answer_dict.get('rule_spec_str'),
+            answer_str=submitted_answer_dict.get('answer_str'))
+
+    def validate(self):
+        """Validates this submitted answer object."""
+        # TODO(msl): These validation methods need tests to ensure that
+        # the right errors show up in the various cases.
+
+        # It's valid for normalized_answer to be None if the interaction type is
+        # Continue.
+        # TODO(bhenning): Validate the normalized answer against future answer
+        # objects after #956 is addressed.
+
+        if self.time_spent_in_sec is None:
+            raise utils.ValidationError(
+                'SubmittedAnswers must have a provided time_spent_in_sec')
+        if self.session_id is None:
+            raise utils.ValidationError(
+                'SubmittedAnswers must have a provided session_id')
+
+        if not isinstance(self.session_id, basestring):
+            raise utils.ValidationError(
+                'Expected session_id to be a string, received %s' %
+                str(self.session_id))
+
+        if not isinstance(self.time_spent_in_sec, float):
```

Comment:
Seems reasonable, though this should never be 0. I think the general chance of it being an integer may be worth fixing this.